programs/fund-sender/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+#![allow(clippy::result_large_err)]
+use crate::utils::errors::ErrorCode;
+use crate::utils::spend::*;
+use crate::utils::state::*;
+use anchor_lang::prelude::*;
+mod utils;
+
+// still need to get a new one
+declare_id!("syriqUnUPcFQjRSaxdFo2wPnXXPjbRsLmhiWUVoGdTo");
+
+#[program]
+pub mod fund_sender {
+    use super::*;
+
+    pub fn register_state(
+        ctx: Context<RegisterState>,
+        sunrise_state: Pubkey,
+        state_in: GenericStateInput,
+    ) -> Result<()> {
+        // register state account on chain, only need to ever be done once
+        let state = &mut ctx.accounts.state;
+        state.sunrise_state = sunrise_state;
+        state.update_authority = state_in.update_authority;
+        state.destination_seed = state_in.destination_seed;
```

Comment:
but that would mean all the program will have the same destination seed? or how can one specify which seed to use inside the program?

Reply:
I will leave it there so that the user can specify it when registering the state such that the same script can be used for deploying a program for different climate projects. Or do you know of a way to allow for that without storing the destination seed in the state?